fe/fe-core/src/main/java/com/starrocks/binlog/BinlogManager.java

```diff
@@ -0,0 +1,273 @@
+        // pass properties not binlogConfig is for
+        // unify the logic of alter table manaul and alter table of mv trigger
+        HashMap<String, String> properties = new HashMap<>();
+        properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_ENABLE, "true");
+        if (!(binlogTtL == INVALID)) {
+            properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_TTL, String.valueOf(binlogTtL));
+        }
+        if (!(binlogMaxSize == INVALID)) {
+            properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_MAX_SIZE, String.valueOf(binlogMaxSize));
+        }
+
+        SchemaChangeHandler schemaChangeHandler = GlobalStateMgr.getCurrentState().getSchemaChangeHandler();
+        return schemaChangeHandler.updateBinlogConfigMeta(db, tableId, properties, TTabletMetaType.BINLOG_CONFIG);
+    }
+
+    // the caller don't need to hold the db lock
+    // for updateBinlogConfigMeta will get the db writelock
+    // return true means that the modification of FEMeta is successful,
+    public boolean tryDisableBinlog(Database db, long tableId) {
+        // modify FE meta, return true
+        // try to distrucite BE task best effort
+        HashMap<String, String> properties = new HashMap<>();
+        properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_ENABLE, "false");
+        SchemaChangeHandler schemaChangeHandler = GlobalStateMgr.getCurrentState().getSchemaChangeHandler();
+        return schemaChangeHandler.updateBinlogConfigMeta(db, tableId, properties, TTabletMetaType.DISABLE_BINLOG);
+    }
+
+
+    public boolean isBinlogAvailable(long dbId, long tableId) {
+        Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+        if (db != null) {
+            try {
+                db.readLock();
+                OlapTable olapTable = (OlapTable) db.getTable(tableId);
+                if (olapTable != null) {
+                    return olapTable.getBinlogAvailableVersion().size() != 0;
+                }
+            } finally {
+                db.readUnlock();
+            }
+        }
+        return false;
+    }
+
+    // isBinlogavailable should be called first to make sure that
+    // the binlog is available
+    // result : partitionId -> binlogAvailableVersion, null indicates the db or table is dropped
+    public Map<Long, Long> getBinlogAvailableVersion(long dbId, long tableId) {
+        HashMap<Partition, Long> partionToAvilableVersionMap = new HashMap<>();
+        Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+        if (db != null) {
+            db.readLock();
+            try {
+                OlapTable olapTable = (OlapTable) db.getTable(tableId);
+                if (olapTable != null) {
+                    return olapTable.getBinlogAvailableVersion();
+                }
+            } finally {
+                db.readUnlock();
+            }
+        }
+        return null;
+    }
+
+    // return all tables with binlog enabled and the latest version itls current binlogConfig
```

Comment:
```suggestion
    // return all tables with binlog enabled and the latest version itls current binlogConfig
```
what itls means? typo?